langchain/src/prompts/tests/few_shot.test.ts

```diff
@@ -1,119 +1,224 @@
+    const partialPrompt = await prompt.partial({
+      foo: () => Promise.resolve("boo"),
+    });
+    expect(await partialPrompt.format({ bar: "baz" })).toBe("boobaz\n");
   });
 
-  const partialPrompt = await prompt.partial({
-    foo: () => Promise.resolve("boo"),
-  });
-  expect(await partialPrompt.format({ bar: "baz" })).toBe("boobaz\n");
-});
+  test("Test partial with function and examples", async () => {
+    const examplePrompt = PromptTemplate.fromTemplate("An example about {x}");
+    const prompt = new FewShotPromptTemplate({
+      prefix: "{foo}{bar}",
+      examples: [{ x: "foo" }, { x: "bar" }],
+      suffix: "",
+      templateFormat: "f-string",
+      exampleSeparator: "\n",
+      examplePrompt,
+      inputVariables: ["foo", "bar"],
+    });
 
-test("Test partial with function and examples", async () => {
-  const examplePrompt = PromptTemplate.fromTemplate("An example about {x}");
-  const prompt = new FewShotPromptTemplate({
-    prefix: "{foo}{bar}",
-    examples: [{ x: "foo" }, { x: "bar" }],
-    suffix: "",
-    templateFormat: "f-string",
-    exampleSeparator: "\n",
-    examplePrompt,
-    inputVariables: ["foo", "bar"],
+    const partialPrompt = await prompt.partial({
+      foo: () => Promise.resolve("boo"),
+    });
+    expect(await partialPrompt.format({ bar: "baz" })).toBe(
+      `boobaz
+An example about foo
+An example about bar
+`
+    );
   });
 
-  const partialPrompt = await prompt.partial({
-    foo: () => Promise.resolve("boo"),
-  });
-  expect(await partialPrompt.format({ bar: "baz" })).toBe(
-    `boobaz
+  test("Test partial with function and example selector", async () => {
+    const examplePrompt = PromptTemplate.fromTemplate("An example about {x}");
+    const exampleSelector = await LengthBasedExampleSelector.fromExamples(
+      [{ x: "foo" }, { x: "bar" }],
+      { examplePrompt, maxLength: 200 }
+    );
+    const prompt = new FewShotPromptTemplate({
+      prefix: "{foo}{bar}",
+      exampleSelector,
+      suffix: "",
+      templateFormat: "f-string",
+      exampleSeparator: "\n",
+      examplePrompt,
+      inputVariables: ["foo", "bar"],
+    });
+
+    const partialPrompt = await prompt.partial({
+      foo: () => Promise.resolve("boo"),
+    });
+    expect(await partialPrompt.format({ bar: "baz" })).toBe(
+      `boobaz
 An example about foo
 An example about bar
 `
-  );
+    );
+  });
 });
 
-test("Test partial with function and example selector", async () => {
-  const examplePrompt = PromptTemplate.fromTemplate("An example about {x}");
-  const exampleSelector = await LengthBasedExampleSelector.fromExamples(
-    [{ x: "foo" }, { x: "bar" }],
-    { examplePrompt, maxLength: 200 }
-  );
-  const prompt = new FewShotPromptTemplate({
-    prefix: "{foo}{bar}",
-    exampleSelector,
-    suffix: "",
-    templateFormat: "f-string",
-    exampleSeparator: "\n",
-    examplePrompt,
-    inputVariables: ["foo", "bar"],
+describe("FewShotChatMessagePromptTemplate", () => {
```

Comment:
The most interesting thing to test here is `formatChatMessages`/`invoke`, which should return something like a list of chat messages I believe?